src/openfl/display/Stage.hx

```diff
@@ -682,13 +677,20 @@ class Stage extends DisplayObjectContainer implements IModule {
 	public function onRenderContextLost (renderer:Renderer):Void {
 		
 		__renderer = null;
+		var stage3D = stage3Ds[0];
+		if (stage3D.context3D != null) {
+			stage3D.__contextRequested = true; // because we want to dispatch `context3DCreate`
+		}
+		stage3D.context3D = null;
```

Comment:
I think I prefer to have something like a `__reset` function in `Stage3D` instead of setting it directly. And because `resize` or `renderStage3D` are always iterating all 3D stages, I think we should do it here as well:

```
for (stage3D in stage.stage3Ds) {
	
	stage3D.__reset ();
	
}
```

Something like this?

Reply:
It is (as usual) weirdly inconsistent in OpenFL in general. E.g. if we look into `GLRenderer` ctor, we'll see that it also only accesses `stage.stage3Ds[0]` directly, so I don't know... As for the reset function, it surely makes sense. Though then again, maybe we should do a similar refactoring in the `GLRenderer` constructor...